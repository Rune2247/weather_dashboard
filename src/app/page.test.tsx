import { render, screen } from "@testing-library/react"
import Home from "./page"

jest.mock("../hooks/useWeather", () => ({
	useWeather: () => ({
		weather: {
			current_weather: {
				weathercode: 2,
				temperature: 20,
				windspeed: 10,
				winddirection: 90
			}
		},
		futureWeather: {
			hourly: {
				time: ["2025-05-29T00:00", "2025-05-29T01:00"],
				precipitation: [0, 0],
				temperature_2m: [20, 21],
				weathercode: [2, 2]
			}
		},
		isLoading: false,
		error: null
	})
}))

describe("Home page", () => {
	it("should render all weather cards", () => {
		render(<Home />)
		expect(screen.getByText("Current Weather")).toBeInTheDocument()
		expect(screen.getByText("Temperature")).toBeInTheDocument()
		expect(screen.getByText("Wind Speed")).toBeInTheDocument()
		expect(screen.getByText("Wind Direction")).toBeInTheDocument()
		expect(screen.getByText("Can we go to crossfit?")).toBeInTheDocument()
	})

	it("should render the ForecastChart if futureWeather is present", () => {
		render(<Home />)
		expect(screen.getByTestId("forecast-chart")).toBeInTheDocument()
	})

	it("should match snapshot", () => {
		const { baseElement } = render(<Home />)
		expect(baseElement).toMatchSnapshot()
	})
})
