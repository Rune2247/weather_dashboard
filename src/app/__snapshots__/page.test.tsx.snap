// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page should match snapshot 1`] = `
<body>
  <div>
    <div
      class="page"
      style="padding: 24px;"
    >
      <div
        style="display: flex; flex-wrap: wrap; justify-content: space-around; gap: 1rem; margin-bottom: 32px; width: 100%;"
      >
        <div
          style="display: flex; align-items: center; background: rgb(255, 255, 255); border-radius: 1rem; box-shadow: 0 2px 12px rgba(0,0,0,0.07); padding: 1.25rem 1.5rem; margin: 0.5rem; min-width: 180px; min-height: 100px; transition: box-shadow 0.2s; gap: 1rem;"
        >
          <div
            style="font-size: 2.2rem; color: rgb(25, 118, 210); display: flex; align-items: center;"
          >
            <svg
              fill="currentColor"
              height="1em"
              id="Layer_1"
              stroke="currentColor"
              stroke-width="0"
              version="1.1"
              viewBox="0 0 30 30"
              width="1em"
              x="0px"
              xmlns="http://www.w3.org/2000/svg"
              y="0px"
            >
              <path
                d="M4.61,16.88c0-1.15,0.36-2.17,1.08-3.07c0.72-0.9,1.63-1.48,2.74-1.73c0.31-1.37,1.02-2.49,2.11-3.37s2.35-1.32,3.76-1.32
	c1.38,0,2.61,0.43,3.69,1.28s1.78,1.95,2.1,3.29h0.33c0.9,0,1.73,0.22,2.49,0.65s1.37,1.03,1.81,1.79c0.44,0.76,0.67,1.58,0.67,2.48
	c0,0.88-0.21,1.7-0.63,2.45s-1,1.35-1.73,1.8c-0.73,0.45-1.54,0.69-2.41,0.72H9.41c-1.34-0.06-2.47-0.57-3.4-1.53
	C5.08,19.37,4.61,18.22,4.61,16.88z M6.32,16.88c0,0.87,0.3,1.62,0.9,2.26s1.33,0.98,2.19,1.03h11.19c0.86-0.04,1.59-0.39,2.19-1.03
	c0.61-0.64,0.91-1.4,0.91-2.26c0-0.88-0.33-1.63-0.98-2.27c-0.65-0.64-1.42-0.96-2.32-0.96H18.8c-0.11,0-0.17-0.06-0.17-0.18
	l-0.07-0.57c-0.11-1.08-0.58-1.99-1.4-2.72c-0.82-0.73-1.77-1.1-2.86-1.1c-1.09,0-2.05,0.37-2.85,1.1
	c-0.81,0.73-1.27,1.64-1.37,2.72l-0.08,0.57c0,0.12-0.07,0.18-0.2,0.18H9.27c-0.84,0.1-1.54,0.46-2.1,1.07S6.32,16.05,6.32,16.88z"
              />
            </svg>
          </div>
          <div
            style="display: flex; flex-direction: column;"
          >
            <div
              style="font-size: 1rem; color: rgb(136, 136, 136); margin-bottom: 4px;"
            >
              Current Weather
            </div>
            <div
              style="font-size: 1.4rem; font-weight: 600; color: rgb(34, 34, 34);"
            >
              Partly cloudy
            </div>
          </div>
        </div>
        <div
          style="display: flex; align-items: center; background: rgb(255, 255, 255); border-radius: 1rem; box-shadow: 0 2px 12px rgba(0,0,0,0.07); padding: 1.25rem 1.5rem; margin: 0.5rem; min-width: 180px; min-height: 100px; transition: box-shadow 0.2s; gap: 1rem;"
        >
          <div
            style="font-size: 2.2rem; color: rgb(25, 118, 210); display: flex; align-items: center;"
          >
            <svg
              fill="currentColor"
              height="1em"
              id="Layer_1"
              stroke="currentColor"
              stroke-width="0"
              version="1.1"
              viewBox="0 0 30 30"
              width="1em"
              x="0px"
              xmlns="http://www.w3.org/2000/svg"
              y="0px"
            >
              <path
                d="M9.91,19.56c0-0.85,0.2-1.64,0.59-2.38s0.94-1.35,1.65-1.84V5.42c0-0.8,0.27-1.48,0.82-2.03S14.2,2.55,15,2.55
	c0.81,0,1.49,0.28,2.04,0.83c0.55,0.56,0.83,1.23,0.83,2.03v9.92c0.71,0.49,1.25,1.11,1.64,1.84s0.58,1.53,0.58,2.38
	c0,0.92-0.23,1.78-0.68,2.56s-1.07,1.4-1.85,1.85s-1.63,0.68-2.56,0.68c-0.92,0-1.77-0.23-2.55-0.68s-1.4-1.07-1.86-1.85
	S9.91,20.48,9.91,19.56z M11.67,19.56c0,0.93,0.33,1.73,0.98,2.39c0.65,0.66,1.44,0.99,2.36,0.99c0.93,0,1.73-0.33,2.4-1
	s1.01-1.46,1.01-2.37c0-0.62-0.16-1.2-0.48-1.73c-0.32-0.53-0.76-0.94-1.32-1.23l-0.28-0.14c-0.1-0.04-0.15-0.14-0.15-0.29V5.42
	c0-0.32-0.11-0.59-0.34-0.81C15.62,4.4,15.34,4.29,15,4.29c-0.32,0-0.6,0.11-0.83,0.32c-0.23,0.21-0.34,0.48-0.34,0.81v10.74
	c0,0.15-0.05,0.25-0.14,0.29l-0.27,0.14c-0.55,0.29-0.98,0.7-1.29,1.23C11.82,18.35,11.67,18.92,11.67,19.56z M12.45,19.56
	c0,0.71,0.24,1.32,0.73,1.82s1.07,0.75,1.76,0.75s1.28-0.25,1.79-0.75c0.51-0.5,0.76-1.11,0.76-1.81c0-0.63-0.22-1.19-0.65-1.67
	c-0.43-0.48-0.96-0.77-1.58-0.85V9.69c0-0.06-0.03-0.13-0.1-0.19c-0.07-0.07-0.14-0.1-0.22-0.1c-0.09,0-0.16,0.03-0.21,0.08
	c-0.05,0.06-0.08,0.12-0.08,0.21v7.34c-0.61,0.09-1.13,0.37-1.56,0.85C12.66,18.37,12.45,18.92,12.45,19.56z"
              />
            </svg>
          </div>
          <div
            style="display: flex; flex-direction: column;"
          >
            <div
              style="font-size: 1rem; color: rgb(136, 136, 136); margin-bottom: 4px;"
            >
              Temperature
            </div>
            <div
              style="font-size: 1.4rem; font-weight: 600; color: rgb(34, 34, 34);"
            >
              20
              <span
                style="font-size: 1rem; color: rgb(25, 118, 210); margin-left: 4px;"
              >
                °C
              </span>
            </div>
          </div>
        </div>
        <div
          style="display: flex; align-items: center; background: rgb(255, 255, 255); border-radius: 1rem; box-shadow: 0 2px 12px rgba(0,0,0,0.07); padding: 1.25rem 1.5rem; margin: 0.5rem; min-width: 180px; min-height: 100px; transition: box-shadow 0.2s; gap: 1rem;"
        >
          <div
            style="font-size: 2.2rem; color: rgb(25, 118, 210); display: flex; align-items: center;"
          >
            <svg
              fill="currentColor"
              height="1em"
              id="Layer_1"
              stroke="currentColor"
              stroke-width="0"
              version="1.1"
              viewBox="0 0 30 30"
              width="1em"
              x="0px"
              xmlns="http://www.w3.org/2000/svg"
              y="0px"
            >
              <path
                d="M3.1,16.97c0,0.24,0.09,0.45,0.28,0.62c0.16,0.19,0.37,0.28,0.63,0.28H18.7c0.29,0,0.53,0.1,0.73,0.3
	c0.2,0.2,0.3,0.45,0.3,0.74c0,0.29-0.1,0.53-0.3,0.72c-0.2,0.19-0.44,0.29-0.74,0.29c-0.29,0-0.54-0.1-0.73-0.29
	c-0.16-0.18-0.36-0.26-0.6-0.26c-0.25,0-0.46,0.09-0.64,0.26s-0.27,0.38-0.27,0.61c0,0.25,0.09,0.46,0.28,0.63
	c0.56,0.55,1.22,0.83,1.96,0.83c0.78,0,1.45-0.27,2.01-0.81c0.56-0.54,0.83-1.19,0.83-1.97s-0.28-1.44-0.84-2
	c-0.56-0.56-1.23-0.84-2-0.84H4.01c-0.25,0-0.46,0.09-0.64,0.26C3.19,16.51,3.1,16.72,3.1,16.97z M3.1,13.69
	c0,0.23,0.09,0.43,0.28,0.61c0.17,0.18,0.38,0.26,0.63,0.26h20.04c0.78,0,1.45-0.27,2.01-0.82c0.56-0.54,0.84-1.2,0.84-1.97
	c0-0.77-0.28-1.44-0.84-1.99s-1.23-0.83-2.01-0.83c-0.77,0-1.42,0.27-1.95,0.8c-0.18,0.16-0.27,0.38-0.27,0.67
	c0,0.26,0.09,0.47,0.26,0.63c0.17,0.16,0.38,0.24,0.63,0.24c0.24,0,0.45-0.08,0.63-0.24c0.19-0.21,0.42-0.31,0.7-0.31
	c0.29,0,0.53,0.1,0.73,0.3c0.2,0.2,0.3,0.44,0.3,0.73c0,0.29-0.1,0.53-0.3,0.72c-0.2,0.19-0.44,0.29-0.73,0.29H4.01
	c-0.25,0-0.46,0.09-0.64,0.26C3.19,13.23,3.1,13.44,3.1,13.69z"
              />
            </svg>
          </div>
          <div
            style="display: flex; flex-direction: column;"
          >
            <div
              style="font-size: 1rem; color: rgb(136, 136, 136); margin-bottom: 4px;"
            >
              Wind Speed
            </div>
            <div
              style="font-size: 1.4rem; font-weight: 600; color: rgb(34, 34, 34);"
            >
              10
              <span
                style="font-size: 1rem; color: rgb(25, 118, 210); margin-left: 4px;"
              >
                km/h
              </span>
            </div>
          </div>
        </div>
        <div
          style="display: flex; align-items: center; background: rgb(255, 255, 255); border-radius: 1rem; box-shadow: 0 2px 12px rgba(0,0,0,0.07); padding: 1.25rem 1.5rem; margin: 0.5rem; min-width: 180px; min-height: 100px; transition: box-shadow 0.2s; gap: 1rem;"
        >
          <div
            style="font-size: 2.2rem; color: rgb(25, 118, 210); display: flex; align-items: center;"
          >
            <svg
              fill="currentColor"
              height="1em"
              id="Layer_1"
              stroke="currentColor"
              stroke-width="0"
              version="1.1"
              viewBox="0 0 30 30"
              width="1em"
              x="0px"
              xmlns="http://www.w3.org/2000/svg"
              y="0px"
            >
              <path
                d="M9.95,10.87c-0.01,0.35,0.1,0.65,0.34,0.9s0.53,0.37,0.89,0.36c0.34,0.02,0.63-0.1,0.88-0.37l1.66-1.64v10.3
	c-0.01,0.35,0.11,0.64,0.36,0.88s0.55,0.35,0.92,0.34c0.34,0.02,0.64-0.09,0.89-0.32s0.39-0.53,0.4-0.88v-10.3l1.64,1.64
	c0.23,0.24,0.53,0.37,0.88,0.37c0.36,0,0.66-0.12,0.9-0.36s0.36-0.53,0.36-0.89c-0.02-0.36-0.15-0.64-0.4-0.85l-3.74-3.84
	c-0.24-0.23-0.55-0.37-0.92-0.4c-0.37,0.02-0.68,0.16-0.92,0.41l-3.75,3.81C10.08,10.25,9.95,10.53,9.95,10.87z"
              />
            </svg>
          </div>
          <div
            style="display: flex; flex-direction: column;"
          >
            <div
              style="font-size: 1rem; color: rgb(136, 136, 136); margin-bottom: 4px;"
            >
              Wind Direction
            </div>
            <div
              style="font-size: 1.4rem; font-weight: 600; color: rgb(34, 34, 34);"
            >
              E (90°)
            </div>
          </div>
        </div>
        <div
          style="display: flex; align-items: center; background: rgb(255, 255, 255); border-radius: 1rem; box-shadow: 0 2px 12px rgba(0,0,0,0.07); padding: 1.25rem 1.5rem; margin: 0.5rem; min-width: 180px; min-height: 100px; transition: box-shadow 0.2s; gap: 1rem;"
        >
          <div
            style="font-size: 2.2rem; color: rgb(25, 118, 210); display: flex; align-items: center;"
          >
            <svg
              fill="currentColor"
              height="1em"
              id="Layer_1"
              stroke="currentColor"
              stroke-width="0"
              version="1.1"
              viewBox="0 0 30 30"
              width="1em"
              x="0px"
              xmlns="http://www.w3.org/2000/svg"
              y="0px"
            >
              <path
                d="M4.37,14.62c0-0.24,0.08-0.45,0.25-0.62c0.17-0.16,0.38-0.24,0.6-0.24h2.04c0.23,0,0.42,0.08,0.58,0.25
	c0.15,0.17,0.23,0.37,0.23,0.61S8,15.06,7.85,15.23c-0.15,0.17-0.35,0.25-0.58,0.25H5.23c-0.23,0-0.43-0.08-0.6-0.25
	C4.46,15.06,4.37,14.86,4.37,14.62z M7.23,21.55c0-0.23,0.08-0.43,0.23-0.61l1.47-1.43c0.15-0.16,0.35-0.23,0.59-0.23
	c0.24,0,0.44,0.08,0.6,0.23s0.24,0.34,0.24,0.57c0,0.24-0.08,0.46-0.24,0.64L8.7,22.14c-0.41,0.32-0.82,0.32-1.23,0
	C7.31,21.98,7.23,21.78,7.23,21.55z M7.23,7.71c0-0.23,0.08-0.43,0.23-0.61C7.66,6.93,7.87,6.85,8.1,6.85
	c0.22,0,0.42,0.08,0.59,0.24l1.43,1.47c0.16,0.15,0.24,0.35,0.24,0.59c0,0.24-0.08,0.44-0.24,0.6s-0.36,0.24-0.6,0.24
	c-0.24,0-0.44-0.08-0.59-0.24L7.47,8.32C7.31,8.16,7.23,7.95,7.23,7.71z M9.78,14.62c0-0.93,0.23-1.8,0.7-2.6s1.1-1.44,1.91-1.91
	s1.67-0.7,2.6-0.7c0.7,0,1.37,0.14,2.02,0.42c0.64,0.28,1.2,0.65,1.66,1.12c0.47,0.47,0.84,1.02,1.11,1.66
	c0.27,0.64,0.41,1.32,0.41,2.02c0,0.94-0.23,1.81-0.7,2.61c-0.47,0.8-1.1,1.43-1.9,1.9c-0.8,0.47-1.67,0.7-2.61,0.7
	s-1.81-0.23-2.61-0.7c-0.8-0.47-1.43-1.1-1.9-1.9C10.02,16.43,9.78,15.56,9.78,14.62z M11.48,14.62c0,0.98,0.34,1.81,1.03,2.5
	c0.68,0.69,1.51,1.04,2.49,1.04s1.81-0.35,2.5-1.04s1.04-1.52,1.04-2.5c0-0.96-0.35-1.78-1.04-2.47c-0.69-0.68-1.52-1.02-2.5-1.02
	c-0.97,0-1.8,0.34-2.48,1.02C11.82,12.84,11.48,13.66,11.48,14.62z M14.14,22.4c0-0.24,0.08-0.44,0.25-0.6s0.37-0.24,0.6-0.24
	c0.24,0,0.45,0.08,0.61,0.24s0.24,0.36,0.24,0.6v1.99c0,0.24-0.08,0.45-0.25,0.62c-0.17,0.17-0.37,0.25-0.6,0.25
	s-0.44-0.08-0.6-0.25c-0.17-0.17-0.25-0.38-0.25-0.62V22.4z M14.14,6.9V4.86c0-0.23,0.08-0.43,0.25-0.6C14.56,4.09,14.76,4,15,4
	s0.43,0.08,0.6,0.25c0.17,0.17,0.25,0.37,0.25,0.6V6.9c0,0.23-0.08,0.42-0.25,0.58S15.23,7.71,15,7.71s-0.44-0.08-0.6-0.23
	S14.14,7.13,14.14,6.9z M19.66,20.08c0-0.23,0.08-0.42,0.23-0.56c0.15-0.16,0.34-0.23,0.56-0.23c0.24,0,0.44,0.08,0.6,0.23
	l1.46,1.43c0.16,0.17,0.24,0.38,0.24,0.61c0,0.23-0.08,0.43-0.24,0.59c-0.4,0.31-0.8,0.31-1.2,0l-1.42-1.42
	C19.74,20.55,19.66,20.34,19.66,20.08z M19.66,9.16c0-0.25,0.08-0.45,0.23-0.59l1.42-1.47c0.17-0.16,0.37-0.24,0.59-0.24
	c0.24,0,0.44,0.08,0.6,0.25c0.17,0.17,0.25,0.37,0.25,0.6c0,0.25-0.08,0.46-0.24,0.62l-1.46,1.43c-0.18,0.16-0.38,0.24-0.6,0.24
	c-0.23,0-0.41-0.08-0.56-0.24S19.66,9.4,19.66,9.16z M21.92,14.62c0-0.24,0.08-0.44,0.24-0.62c0.16-0.16,0.35-0.24,0.57-0.24h2.02
	c0.23,0,0.43,0.09,0.6,0.26c0.17,0.17,0.26,0.37,0.26,0.6s-0.09,0.43-0.26,0.6c-0.17,0.17-0.37,0.25-0.6,0.25h-2.02
	c-0.23,0-0.43-0.08-0.58-0.25S21.92,14.86,21.92,14.62z"
              />
            </svg>
          </div>
          <div
            style="display: flex; flex-direction: column;"
          >
            <div
              style="font-size: 1rem; color: rgb(136, 136, 136); margin-bottom: 4px;"
            >
              Can we go to crossfit?
            </div>
            <div
              style="font-size: 1.4rem; font-weight: 600; color: rgb(34, 34, 34);"
            >
              Yes, go for it!
            </div>
          </div>
        </div>
      </div>
      <div
        data-testid="forecast-chart"
        style="width: 100%; max-width: 1100px; height: 340px; margin: 32px auto 0px auto; padding: 32px; background: rgb(255, 255, 255); border-radius: 1rem; box-shadow: 0 2px 12px rgba(0,0,0,0.07);"
      >
        <div
          class="recharts-responsive-container"
          style="width: 100%; height: 100%; min-width: 0;"
        />
      </div>
    </div>
  </div>
</body>
`;
